ChocolateDelightsDB/3 reports.sql

```diff
@@ -0,0 +1,59 @@
+use ChocolateDelightsDB
+go
+
+--1. The amount we sell of each product (in order of most popular to least popular)
+select o.ItemDesc, QuantitySold = sum(o.Quantity)
+from Orders o
+group by o.ItemDesc
+order by QuantitySold desc
+--2. Product details (item number, description, cost of production, wholesale price)
+select distinct o.ItemNum, o.ItemDesc, o.ProductionCost, o.WholesalePrice
+from orders o
+--3. Which products brought in the most money
+select top(1) o.ItemDesc, Profit = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity)
+from orders o
+group by o.ItemDesc
+order by Profit desc
+--4. Which product brought in the least money
+select top(1) o.ItemDesc, Profit = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity)
+from orders o
+group by o.ItemDesc
+order by Profit 
+--5. The peak month for each product
+select o.ItemDesc, o.MonthPurchased, Profit = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity)
+from orders o
+group by o.ItemDesc, o.MonthPurchased
+order by Profit desc
+--6. Which company brings in the most money
+select top(1) o.Company, Profit = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity)
+from orders o
+group by o.Company
+order by Profit desc
+--7. Which customer brings in the least  money
+select top(1) o.Company, Profit = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity)
+from orders o
+group by o.Company
+order by Profit
+--8. How much profit each customer makes on our products (assuming they sell everything)
+select o.Company, Profit = sum((o.RetailPrice - o.WholesalePrice) * o.Quantity)
+from orders o
+group by o.Company
+order by Profit
+--6. How profit increased/decreased by year per item
+; 
+with x as(
+	select o.YearPurchased, o.ItemDesc, Profit = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity)
+	from Orders o
+	where o.YearPurchased = 2020
+	group by o.YearPurchased, o.ItemDesc
+)
+select o.ItemDesc, ProfitChange = sum((o.WholesalePrice - o.ProductionCost) * o.Quantity) - x.Profit
+from Orders o
+join x
+on o.ItemDesc = x.ItemDesc
+where o.YearPurchased = 2021
+group by o.YearPurchased, o.ItemDesc, x.Profit
+order by o.YearPurchased
+
```

Comment:
This report did not run any data. Please fix

Reply:
I think that it does run data if there is data that matches the criteria.  It could be it's not returning any data because there are no products which were sold 2 consecutive years in a row.  I put in some test data and it does seem to work.  Please look into this and let me know if it needs to be fixed, or if it is working once more data is inserted

Reply:
Why did you put the years in as 2020 and 2021? I want the difference in each year. You can have the profits side by side from each year.

Reply:
That's not something I know how to do offhand, I'm looking into it, it might not be in the scope of this course, I can let you know.  If you know of a way to do this, I'm happy for a tip

Reply:
Can you provide me with some more data?  None of the sample data works for this report, I need products sold in consecutive years in order to do this report.
Thanks, I think I had misunderstood and I can try to do it in one select statement but if it's not working, I can do a union select

Reply:
No problem, yes please.  I just wanted some more data of products bought in consecutive years, so that this report will return data.  For example, chocolate hearts sold in 2018 and 2019 etc. Thanks!

Reply:
Hi I added some more data. Let me know if you're able to get it.

Reply:
Thank you
Where?  Did you push it to this repo?

Reply:
I tried to push but it wasn't working. What do I do?

Reply:
I pushed the data into the homework session.

Reply:
How do I push directly into here?

Reply:
okay, it's fine, got it, thanks

Reply:
I'm so sorry for the delay, I was out for a couple days.  Please look at my work, I hope all is good to go

Reply:
We can close it once everything has been approved by you

Reply:
Thank you for all your work! I am closing the pull request